services/user/Subgroups/service/src/lib.rs

```diff
@@ -1,98 +1,72 @@
 #[psibase::service(name = "subgroups")]
 pub mod service {
     pub use psibase::fracpack;
-    use std::collections::HashMap;
 
-    /// # Description
-    ///
-    /// Recursive function that finds a valid partition of `n` into groups with sizes from `allowed`
-    /// Uses memoization to avoid redundant calculations
+    /// Function that determines which subgrouping solution is preferred based on specific criteria.
+    /// Returns true if the new solution is preferred over the old solution.
+    type PreferenceFunction = fn(new: &[u32], old: &[u32]) -> bool;
+
+    /// Preference function that compares based on the greatest minimum group size.
+    /// If minimum values are equal, it compares based on maximum group size.
```

Comment:
What about sorting in ascending order and then comparing lexicographically.

Reply:
Which has the same sort order as a tuple of the counts of groups of each size.

Reply:
Elegant